### PR TITLE
v1.15 Backports 2024-01-30

### PR DESCRIPTION
**Pull request by joamaki**
* [x] #30340 (@dylandreimerink)
 * [x] #30286 (@julianwiedmann)
 * [x] #30154 (@ldelossa)
 * [x] #30282 (@giorio94)
 * [x] #30167 (@viktor-kurchenko)
 * [x] #29423 (@derailed)
 * [x] #29938 (@M0NsTeRRR)
 * [x] #30220 (@danehans)
 * [x] #30299 (@rgo3)
 * [x] #30403 (@f1ko)
 * [x] #30052 (@yingnanzhang666)
 * [x] #30410 (@julianwiedmann)
 * [x] #30399 (@tlcowling)
 * [x] #30219 (@dependabot[bot])
 * [x] #30236 (@soggiest)
 * [x] #30407 (@nvibert)
 * [x] #30445 (@meyskens)
 * [x] #30477 (@sayboras)
 * [x] #30488 (@julianwiedmann)
 * [x] #30422 (@geakstr :warning: regenerated helm template etc.)
 * [x] #30442 (@julianwiedmann)
 * [x] #30365 (@tommyp1ckles)
 * [x] #30503 (@qmonnet)
 * [x] #30416 (@joamaki :warning: adapted the renaming)
 * [x] #30329 (@3u13r)
 * [x] #30483 (@sayboras)
 * [x] #30504 (@marseel)
 * [x] #30247 (@marseel :warning: fixed up CODEOWNERS)

PRs skipped due to conflicts:

 * #30337 (@harsimran-pabla)
 * #30100 (@chaunceyjiang)
 * #30443 (@brlbil)

Once this PR is merged, a GitHub action will update the labels of these PRs:
```upstream-prs
 29025 29023 29024 30340 30286 30154 30282 30167 29423 29938 30220 30299 30403 30052 30410 30399 30219 30236 30407 30445 30477 30488 30422 30442 30365 30503 30416 30329 30483 30504 30247
```

### CODE REVIEW

**Review by giorio94 — APPROVED** [review]
My commit looks good, thanks!

**Review by qmonnet — APPROVED** [review]

**Review by geakstr — APPROVED** [review]

**Comment by aanm** [comment]
/test-backport-1.15

**Review by dylandreimerink — APPROVED** [review]

**Review by marseel — APPROVED** [review]

**Review by viktor-kurchenko — APPROVED** [review]
LGTM, Thank you!

**Comment by nvibert** [comment]
LGTM !

**Review by derailed — APPROVED** [review]
@joamaki LGTM Thanks for these updates Jussi!

**Review by joestringer — APPROVED** [review]

**Comment by joestringer** [comment]
(All tests passed. docs-builder triggered an update on top)

**Review by rgo3 — APPROVED** [review]

**Comment by joestringer** [comment]
/test-backport-1.15

**Review by sayboras — APPROVED** [review]
Thanks and looks good for my commits.

**Review by ldelossa — APPROVED** [review]
Changes LGTM

File: CODEOWNERS
Comment (joestringer) [inline CODEOWNERS:224]: I don't think we need to block the PR on this but I wanted to point out to @marseel / @cilium/sig-scalability that in general we expect most groups to apply their review feedback during the primary PR against `main` then the backport is a sanity check, so for this case @cilium/sig-scalability does not need to feel obliged to review all changes to stable branch modifications of this workflow. Perhaps as a followup we can just drop this change and defer to the existing rule.

```suggestion
```

**Review by tommyp1ckles — APPROVED** [review]

**Review by julianwiedmann — APPROVED** [review]

File: CODEOWNERS
Comment (marseel) [reply CODEOWNERS:224]: Yes, that is totally fine with me - modification of CODEOWNERS was part of the commit where I introduced a new test, which seems reasonable in general, but I guess for backports we should get rid of it. I will open followup PR.

**Comment by joamaki** [comment]
> @joamaki Looks like the first three PRs listed in the description are not included in this backport?

Looks like they were in v1.15 already. Fixed the labels.

**Review by aanm — APPROVED** [review]
I've reviewed the remaining backported PRs.

**Pull request merged**
